### DISCUSSION TITLE
Remove events in processor

### DISCUSSION THREAD

**Discussion by Tobbeman** [open-telemetry/opentelemetry-dotnet#6006] [Q&A]
Hello,

We are using Pomelo, which in turn is using MySqlConnector
MySqlConnector is generating very nice activities for us to observe, however these activities have events which we would like to filter out since AzureMonitor sends as [traces](https://github.com/Azure/azure-sdk-for-net/blob/e986729b7f6d1367ceba27e8dd31a685d9ad65a8/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs#L250), increasing our cost.

We have a processor today, but would like to extend it to also remove events.
Is this achievable somehow? Activity.Events seems to be read only 

Our current processor:
```c#
internal class OpenTelemetryFilterProcessor : BaseProcessor<Activity>
{
    public override void OnStart(Activity activity)
    {
        FilterPomeloMySqlConnector(activity);
    }

    void FilterPomeloMySqlConnector(Activity activity) 
    {
        if (activity.Source.Name != "MySqlConnector") 
        {
            return;
        }

        // TL: Really noisy, add little value
        // One is created for every query
        if (activity.DisplayName == "Open")
        {
            activity.IsAllDataRequested = false;
        }
    }
}
```

**Comment by cijothomas** [comment]
Not possible as there is no spec about removing SpanEvents (ActivityEvents...).  I think the best option is to request AzureMonitor to provide a feature flag to *not* export ActivityEvents. Or better - report the issue at the source, and request the instrumentation to provide an opt-out feature from ActivityEvents.
Such an issue was previously reported, and the instrumentation offered a opt-out mechanism - https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.StackExchangeRedis/StackExchangeRedisInstrumentationOptions.cs#L35
